web/pandas/pdeps/0005-no-default-index-mode.md

```diff
@@ -0,0 +1,189 @@
+# PDEP-5: No-default-index mode
+
+- Created: 14 November 2022
+- Status: Draft
+- Discussion: [#49693](https://github.com/pandas-dev/pandas/pull/49693)
+- Author: [Marco Gorelli](https://github.com/MarcoGorelli)
+- Revision: 1
+
+## Abstract
+
+The suggestion is to add a `mode.no_default_index` option which, if enabled,
+would ensure:
+- if a ``DataFrame`` / ``Series`` is created, then by default it won't have an ``Index``;
+- nobody will get an ``Index`` unless they ask for one - this would affect the default behaviour of ``groupby``, ``value_counts``, ``pivot_table``, and more.
+
+This option would not be the default. Users would need to explicitly opt-in to it, via ``pd.set_option('mode.no_default_index', True)``, via ``pd.option_context``, or via the ``PANDAS_NO_DEFAULT_INDEX`` environment variable.
+
+## Motivation and Scope
+
+The Index can be a source of confusion and frustration for pandas users. For example, let's consider the inputs
+
+```python
+In [37]: ser1 = df.groupby('sender')['amount'].sum()
```

Comment:
What is the output of a groupby when no_index is set?

Reply:
It'd be the same as with `as_index=False`